### PR TITLE
[Compression] preview doc

### PR DESCRIPTION
**Pull request by J-shang**
#### Test Options ####
  - [ ] fast test
  - [ ] full test - HPO
  - [ ] full test - NAS
  - [ ] full test - compression

### Checklist ###
  - [ ] test case
  - [ ] doc

### CODE REVIEW

File: docs/source/compression_preview/changes.rst [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+Nonetheless, if you have employed NNI Compression before and want to try the latest Compression version,
+this document will help you in comprehending the noteworthy alterations in the interface in 3.0.
+
+
+New compression version import path:
+
+.. code-block:: python
+
+    # most new compression related, include pruners, quantizers, distillers, except new pruning speedup
+    from nni.contrib.compression.xxx import xxx
```
Comment (QuanluZhang) [inline docs/source/compression_preview/changes.rst:16]: still under `contrib`?

File: docs/source/compression_preview/changes.rst [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+In the previous version of NNI (lower than 3.0), three pruning modes were supported: ``normal``, ``global``, and ``dependency-aware``.
+
+In the ``normal`` mode, each module was required to be assigned a sparse ratio, and the pruner generated masks directly on the weight elements of this ratio.
+
+In the ``global`` mode, a sparse ratio was set for a group of modules, and the pruner generated masks whose overall sparse ratio conformed to the setting,
+but the sparsity of each module in the group may differ.
+
+The ``dependency-aware`` mode constrained modules with operational dependencies to generate related masks.
+
```
Comment (QuanluZhang) [inline docs/source/compression_preview/changes.rst:64]: remove this line?

File: docs/source/compression_preview/changes.rst (hunk 1 above)
Comment (J-shang) [reply docs/source/compression_preview/changes.rst:16]: Yes, according to the previous discussion, new compression is in contrib in 3.0 preview, and will replace it in the official version.

File: docs/source/compression_preview/changes.rst [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+Right now in NNI 3.0, users can directly set :ref:`global_group_id` and :ref:`dependency_group_id` to implement ``global`` and ``dependency-aware`` modes.
+Additionally, :ref:`align` is supported to generate a mask from another module mask, such as generating a batch normalization mask from a convolution mask.
+You can achieve improved performance and exploration by combining these modes by setting the appropriate keys in the configuration list.
+
+
+Pruning Speedup
+---------------
+
+The previous pruning speedup is based on ``torch.jit.trace`` to trace the model graph,
+it has a lot limiatations and need additional support for performing the operations.
```
Comment (QuanluZhang) [inline docs/source/compression_preview/changes.rst:81]: limitations

Comment (QuanluZhang) [reply docs/source/compression_preview/changes.rst:81]: needs

File: docs/source/compression_preview/changes.rst [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+Most operations that cannot be traced in previous pruning speedup can now be traced.
+
+In addition to ``concrete_trace``, if users already have a good ``torch.fx.GraphModule`` for their model traced,
+users can also use the ``torch.fx.GraphModule`` directly.
+
+Moreover, customized masks propagation logic and module replacement method are supported in new pruning speedup.
+
+The previous method of pruning speedup relied on ``torch.jit.trace`` to trace the model graph.
+However, this method had several limitations and required additional support to perform certain operations.
+These limitations resulted in excessive maintenance costs, making it difficult to continue development.
```
Comment (QuanluZhang) [inline docs/source/compression_preview/changes.rst:96]: appears again?

File: docs/source/compression_preview/config_list.rst [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+``default``, ``in_channel``, ``out_channel``, ``per_channel`` and list of integer are supported:
+
+- default: The pruner will auto determine using which kind of granularity, usually consistent with the paper.
+- in_channel: The pruner will do pruning on the weight parameters 1 dimension.
+- out_channel: The pruner will do pruning on the weight parameters 0 dimension.
+- per_channel: The pruner will do pruning on the input/output -1 dimension.
+- list of integer: Block sparse will be applied. For example, ``[4, 4]`` will apply 4x4 block sparse on the last two dimensions of the weight parameters.
+
+Note that ``in_channel`` or ``out_channel`` is not supported for input/output targets, please using ``per_channel`` instead.
+``torch.nn.Embedding`` is special, it's output dimension on weight is 1, so if want to pruning Embedding output channel, please set ``in_channel`` for its granularity for workaround.
```
Comment (QuanluZhang) [inline docs/source/compression_preview/config_list.rst:196]: why is there a special case for embedding?

File: docs/source/compression_preview/config_list.rst [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+
+link
+^^^^
+
+A teacher module name or a list of teacher module names. The student module link to.
+
+apply_method
+^^^^^^^^^^^^
+
+``mse`` or ``kl``.
```
Comment (QuanluZhang) [inline docs/source/compression_preview/config_list.rst:311]: may explain ``kl`` here

File: docs/source/compression_preview/setting.rst [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+    customized_setting = {
+        '_input_': {
+            'quant_dtype': None,
+            'quant_scheme': None,
+            'granularity': [-1, 1],
+            'apply_method': 'clamp_round',
+        },
+        'p1': {
+            'quant_dtype': None,
+            'quant_scheme': None,
```
Comment (QuanluZhang) [inline docs/source/compression_preview/setting.rst:152]: what is the meaning of `None` for `quant_dtype` and `quant_scheme`?

File: docs/source/compression_preview/setting.rst [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+
+    QuantizationSetting.register('CustomizedModule', customized_setting)
+
+Temporarily Setting Update
+--------------------------
+
+Sometimes we just want to temporarily modify the setting template and don't want to make a global change.
+Then we could directly write full setting in ``config_list`` to achieve this.
+
+For example, if the compressed model has conv-bn-relu pattern, and for a better pruning simulation and performence,
```
Comment (QuanluZhang) [inline docs/source/compression_preview/setting.rst:178]: performance

File: docs/source/compression_preview/setting.rst [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+            'quant_dtype': None,
+            'quant_scheme': None,
+            'granularity': [-1, 1],
+            'apply_method': 'clamp_round',
+        }
+    }
+
+    QuantizationSetting.register('CustomizedModule', customized_setting)
+
+Temporarily Setting Update
```
Comment (QuanluZhang) [inline docs/source/compression_preview/setting.rst:172]: what is the meaning of "temporarily"?

File: docs/source/compression_preview/setting.rst (hunk 7 above)
Comment (J-shang) [reply docs/source/compression_preview/setting.rst:152]: `None` means has no default value, it a template setting for this customized module type, not a config, that means when user writing a config, `quant_dtype` and `quant_scheme` should be configured.

File: docs/source/compression_preview/config_list.rst (hunk 5 above)
Comment (J-shang) [reply docs/source/compression_preview/config_list.rst:196]: because most layer output related to weight dim 0, but embedding is related to weight dim 1.

File: docs/source/compression_preview/changes.rst (hunk 4 above)
Comment (J-shang) [reply docs/source/compression_preview/changes.rst:96]: forgot to remove, removed

File: docs/source/compression_preview/changes.rst (hunk 2 above)
Comment (J-shang) [reply docs/source/compression_preview/changes.rst:64]: why?

File: docs/source/compression_preview/setting.rst (hunk 9 above)
Comment (J-shang) [reply docs/source/compression_preview/setting.rst:172]: `temporarily` means this update only takes effect in the current subconfig. In default setting template, BatchNorm2d should generate masks by pruning algorithm, and example in this section, `weight` mask generation for `bn1` and `bn2` is align to the `conv1` or `conv2` weight masks, not generate by pruning algorithm. But this generation change is a temporarily change, that means other BatchNorm2d will still follow the default setting template.

File: docs/source/compression_preview/config_list.rst (hunk 6 above)
Comment (J-shang) [reply docs/source/compression_preview/config_list.rst:311]: fixed

**Review by QuanluZhang — APPROVED** [review]

**Review by Bonytu — APPROVED** [review]
